install/upgrade.php

```diff
@@ -222,6 +225,24 @@ protected function upgrade_3_7() {
 			'pll_language_from_content_available',
 			0 === $this->options['force_lang'] ? 'yes' : 'no'
 		);
+
+		$options = new Options();
+		$model   = new PLL_Model( $options );
+
+		foreach ( $model->get_languages_list() as $language ) {
+			$mo = new PLL_MO();
+			$mo->import_from_db( $language );
+
+			foreach ( $mo->entries as $entry ) {
+				if ( $entry->singular === $entry->translations[0] ) {
+					$entry->translations = array( '' );
+				}
+
+				$mo->add_entry_or_merge( $entry );
+			}
+
+			$mo->export_to_db( $language );
+		}
```

Comment:
1. To clarify and avoid mixing unrelated things, I would separate `upgrade_3_7()` in 2 methods as we did for 3.4.
2. I would manage the upgrade at a lower level (dealing directly with the meta). This is what I always did in the past. The reasoning is that, once merged, we will *never* touch this code anymore. While other parts of Polylang will continue to evolve and this *may* break this. I want to be sure that the upgrade process will continue to work in far future.

Reply:
👍 
See 9c1f1d8e